packages/react/src/primitives/SelectField/SelectField.tsx

```diff
@@ -10,6 +10,31 @@ import { SelectFieldProps, Primitive } from '../types';
 import { splitPrimitiveProps } from '../shared/styleUtils';
 import { useStableId } from '../shared/utils';
 
+interface SelectFieldChildrenProps {
+  children?: React.ReactNode;
+  options?: SelectFieldProps['options'];
+}
+
+const selectFieldChildren = ({
+  children,
+  options,
+}: SelectFieldChildrenProps) => {
+  if (children) {
+    if (options?.length) {
+      console.warn(
+        'Amplify UI: <SelectField> component  defaults to rendering children over `options`. When using the `options` prop, omit children.'
+      );
+    }
+    return children;
+  }
+
+  return options?.map((option, index) => (
+    <option label={option} value={option} key={`${option}-${index}`}>
```

Comment:
Just think out loud, can we just use the `option` as key? Appending index will somehow affect React's optimization. Like if I have an option `<option label="apple" value="apple" key={'apple-2'}>` at index 2, but after updates, I got the same option `<option label="apple" value="apple" key={'apple-1'}>` move to index 1. However, React will think they are different because their keys are affected by the index and rerender it versus reuse the original one

Reply:
Good reading <- https://reactjs.org/docs/reconciliation.html